Makefile

```diff
@@ -377,8 +378,26 @@ test:
 	fi; \
 	PYTHONDONTWRITEBYTECODE=1 py.test -p no:cacheprovider -n auto $(TEST_DIRS)
 	cd awxkit && $(VENV_BASE)/awx/bin/tox -re py2,py3
+	cd /awx_devel
+	make prepare_test_modules
 	awx-manage check_migrations --dry-run --check  -n 'vNNN_missing_migration_file'
 
+prepare_modules_venv:
+	rm -rf $(MODULES_VENV)
+	mkdir $(MODULES_VENV)
+	ln -s /usr/lib/python2.7/site-packages/ansible $(MODULES_VENV)/ansible
+	$(VENV_BASE)/awx/bin/pip install --target=$(MODULES_VENV) git+https://github.com/ansible/tower-cli.git
+
+MODULES_TEST_DIRS ?= awx_modules/test/awx
+
+test_modules:
+	@if [ "$(VENV_BASE)" ]; then \
+		. $(VENV_BASE)/awx/bin/activate; \
+	fi; \
+	PYTHONPATH=$(MODULES_VENV):/awx_devel/awx_modules:$PYTHONPATH py.test $(MODULES_TEST_DIRS)
+
+prepare_test_modules: prepare_modules_venv test_modules
```

Comment:
This is confusing to me, the target is named prepare but it is actually running the tests. I would suggest moving the `prepare_modules_venv` as a dependency of test_modules and drop this target entirely.

Reply:
Due to path issues, I still strongly want some form of:

```
make test_modules MODULE_TEST_DIRS=awx_modules/test/awx/test_organization.py
```

to be available for development.

Adding the install and flake8 step to this target doesn't work with that goal to do quick single-test runs.

The naming of the combined target is complicated by the reference from https://github.com/ansible/zuul-jobs/pull/35

Reply:
So this is needed for testing the modules isolated, right?

Reply:
Talked offline with @AlanCoding and he explained that the intetion to have 3 make targets are:

* To be able to run only the prepare environment
* To be able to run only the tests for the modules
* To do both, prepare the environment and run the test with a single make target

Initially I thought having those 3 wouldn't be necessary but we actually need the three targets.